index.bs

```diff
@@ -3497,6 +3652,241 @@ partial dictionary MLOpSupportLimits {
+        builder.greater(input, zero),
+        positiveOne,
+        builder.where(
+          builder.lesser(input, zero),
+          negativeOne,
+          zero));
+    }
+    </pre>
+  </details>
+</div>
+
+
+### dequantizeLinear ### {#api-mlgraphbuilder-dequantizelinear}
+Dequantizes an integer tensor to floating point space using the scale and zero-point bias, where `output = (input - zeroPoint) * scale`.
+
+TODO: Elaborate on blockwise broadcasting - The operation will be [=broadcast=] according to [[!numpy-broadcasting-rule]]. The input tensors must be [=bidirectionally broadcastable=]. The [=MLOperand/rank=] of the output tensor is the maximum [=MLOperand/rank=] of the input tensors. For each dimension of the output tensor, its size is the maximum size along that dimension of the input tensors, and each dimension must be blockwise compatible with the output (e.g. given an input shape *[12]*, scales of the following shapes are blockwise compatible {*[1]*, *[3]*, *[4]*, *[6]*, *[12]*} as they are all multiples of the input dimensions, but a shape of *[5]* would not be).
+
+<script type=idl>
+partial interface MLGraphBuilder {
+  MLOperand dequantizeLinear(MLOperand input,
+                             MLOperand scale,
+                             MLOperand zeroPoint,
+                             optional MLOperatorOptions options = {});
+};
+
+dictionary MLQuantizationSupportLimits {
+  MLSupportLimits input;
+  MLSupportLimits scale;
+  MLSupportLimits zeroPoint;
+  MLSupportLimits output;
+};
+
+partial dictionary MLOpSupportLimits {
+  MLQuantizationSupportLimits dequantizeLinear;
+};
+</script>
+
+<div dfn-for="MLGraphBuilder/dequantizeLinear(input, scale, zeroPoint, options)" dfn-type=argument>
+    **Arguments:**
+        - <dfn>input</dfn>: an {{MLOperand}}. The input tensor.
+        - <dfn>scale</dfn>: an {{MLOperand}}. The scale tensor to multiply each input value by after adjusting by the zero point.
+        - <dfn>zeroPoint</dfn>: an {{MLOperand}}. The zero point tensor to subtract from each input value.
+        - <dfn>options</dfn>: an {{MLOperatorOptions}}. Specifies the optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output tensor that contains the dequantized values.
+</div>
+
+<table id=constraints-dequantizelinear class='data' link-for="MLGraphBuilder/dequantizeLinear(input, scale, zeroPoint, options)">
+  <caption>Constraints for {{MLGraphBuilder/dequantizeLinear()}}</caption>
+  <thead>
+    <tr>
+      <th>operand</th>
+      <th>[=/allowed data types=]</th>
+      <th>[=/allowed ranks=]</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>{{input}}</td>
+    <td>{{MLOperandDataType/"uint4"}}, {{MLOperandDataType/"uint8"}}, {{MLOperandDataType/"uint32"}}, {{MLOperandDataType/"int32"}}</td>
+    <td>N</td>
+  </tr>
+  <tr>
+    <td>{{scale}}</td>
+    <td>{{MLOperandDataType/"float32"}}, {{MLOperandDataType/"float16"}}</td>
+    <td>0 to {{input}}'s [=MLOperand/rank=]</td>
```

Comment:
To reduce testing complexity some, we could require the scale and zeroPoint are explicitly reshaped to the same rank as input ahead of time? That would also nicely resolve variant oddities in how the caller express axes ahead of time🤔.

Reply:
Reviewing other ops, requiring scale and zeroPoint to be the same rank would be consistent. 

If someone was manually coding against the API then supporting scalars would be convenient, but I don't think we should prioritize that.